src/common/main/kotlin/tech/thatgravyboat/skyblockapi/api/profile/garden/PlotsAPI.kt

```diff
@@ -0,0 +1,308 @@
+    )
+
+    private val chatSingularSpawnRegex = chatGroup.create(
+        "singular_spawn",
+        "[^:]*! A Pest has appeared in Plot - (?<name>.+)!",
+    )
+    private val chatPluralSpawnRegex = chatGroup.create(
+        "plural_spawn",
+        "[^:]+! (?<amount>\\d+) Pests have spawned in Plot - (?<name>.+)!",
+    )
+    private val chatOfflineSpawnRegex = chatGroup.create(
+        "offline",
+        "[^:]+! While you were offline, Pests spawned in Plots (?<plots>.*)!",
+    )
+
+    //endregion
+
+    private var currentPlotSlot = 1
+
+    val plots = listOf(
+        listOf(21, 13, 9, 14, 22),
+        listOf(15, 5, 1, 6, 16),
+        listOf(10, 2, 0, 3, 11),
+        listOf(17, 7, 4, 8, 18),
+        listOf(23, 19, 12, 20, 24),
+    ).withIndex().flatMap { (z, row) ->
+        row.withIndex().map { (x, id) ->
+            val minX = x * 96 - 240.0
+            val minZ = z * 96 - 240.0
+            val maxX = x * 96 - 144.0
+            val maxZ = z * 96 - 144.0
+            val aabb = AABB(minX, 0.0, minZ, maxX, 256.0, maxZ)
+            currentPlotSlot++
+            Plot(id, currentPlotSlot, aabb)
+        }.also { currentPlotSlot += 4 }
+    }
+
+    var currentPestAmount = 0
+        private set
+
+
+    fun getPlot(id: Int): Plot? = plots.find { it.id == id }
+    fun getPlotByName(name: String): Plot? = plots.find { it.data?.name == name }
+
+    fun getCurrentPlot(): Plot? = plots.find { McPlayer in it.aabb }
+
+
+    private fun clearPests() {
+        currentPestAmount = 0
+        plots.forEach { plot ->
+            plot.data?.pest = Pest(0, inaccurate = false)
+            plot.data?.save()
+        }
+    }
+
+    @Subscription
+    @OnlyIn(SkyBlockIsland.GARDEN)
+    fun onScoreboardUpdate(event: ScoreboardUpdateEvent) {
+        if (scoreboardNoPestsRegex.anyMatch(event.new)) {
+            clearPests()
+            return
+        }
+
+        scoreboardPestAmountRegex.anyMatch(event.new, "amount") { (amount) ->
+            currentPestAmount = amount.toIntValue()
+        }
+
+        scoreboardPlotPestAmountRegex.anyMatch(event.new, "name", "amount") { (name, amount) ->
+            val plot = plots.find { it.data?.name == name } ?: return@anyMatch
+            val pest = Pest(amount.toIntValue(), inaccurate = false)
+            plot.data?.pest = pest
+            plot.data?.save()
+        }
+    }
+
+    @Subscription
+    @OnlyIn(SkyBlockIsland.GARDEN)
+    @MustBeContainer
+    @InventoryTitle("Configure Plots")
+    private fun InventoryChangeEvent.onInventoryChange() {
+        val plotId = plots.find { it.slot == slot.index }?.id ?: return
+        val itemId = item.getSkyBlockId() ?: item.getItemModel().takeUnless { it == Items.OAK_BUTTON || it in ItemTag.GLASS_PANES }?.let {
+            BuiltInRegistries.ITEM.getKey(it.asItem()).toString()
+        }
+
+        var name = ""
+        deskPlotNameRegex.match(item.cleanName, "name") { (plotName) ->
+            name = plotName
+        }
```

Comment:
In RegexUtils add matchOrNull like findOrNull that already exists and then replace this with the following.

```suggestion
        var name = deskPlotNameRegex.matchOrNull(item.cleanName, "name", Destructured::component1) ?: ""
```

Reply:
im pretty sure i added a function thats just "matchGroup"

Reply:
oh wait the pr is already merged

Reply:
you had findGroup, I forgot about it shouldve probs added matchGroup instead but doesnt matter that much

Reply:
ah ok